packages/@expo/prebuild-config/src/plugins/unversioned/react-native-edge-to-edge/helpers.ts

```diff
@@ -0,0 +1,39 @@
+import type { ExpoConfig } from '@expo/config-types';
+
+import { EdgeToEdgePlugin } from './withEdgeToEdge';
+
+export function edgeToEdgePluginIndex(config: ExpoConfig): number | null {
+  const noArgumentPluginIndex =
+    config.plugins?.findIndex(
+      (plugin) => typeof plugin === 'string' && plugin.includes('react-native-edge-to-edge')
+    ) ?? -1;
+
+  const argumentPluginIndex =
+    config.plugins?.findIndex(
+      (plugin) =>
+        Array.isArray(plugin) &&
+        typeof plugin[0] === 'string' &&
+        plugin[0].includes('react-native-edge-to-edge')
+    ) ?? -1;
+
+  const pluginIndex = Math.max(noArgumentPluginIndex, argumentPluginIndex);
+
+  if (pluginIndex === -1) {
+    return null;
+  }
+  return pluginIndex;
+}
+
+export function hasEnabledEdgeToEdge(config: ExpoConfig) {
+  return config.android?.edgeToEdgeEnabled === true || edgeToEdgePluginIndex(config) != null;
+}
+
+export function loadEdgeToEdgeConfigPlugin(): EdgeToEdgePlugin | null {
+  try {
+    // @ts-ignore <-- edge-to-edge plugin doesn't export a type definition
+    const { default: plugin } = require('react-native-edge-to-edge/app.plugin');
```

Comment:
we should try both
`projectRoot -> expo -> react-native-edge-to-edge` (transitive dep)
`projectRoot -> react-native-edge-to-edge` (people explicit direct dep)

fwiw, in pnpm isolation mode, the `require('react-native-edge-to-edge/app.plugin');` may not work for the transitive dep

you can use something like
```ts
const path = resolveFrom.silent();
if (path) {
  const { default: plugin } = require(path);
}
```